Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -260,18 +277,30 @@ private Stream CacheAndCreateEntryStream(string filename, string entryName)
         /// <param name="entryName">The name of the entry</param>
         /// <param name="fileName">The name of the zip file on disk</param>
         /// <returns>A <see cref="Stream"/> of the appropriate zip entry</returns>
-        private static Stream CreateEntryStream(CachedZipFile zipFile, string entryName, string fileName)
+        private Stream CreateEntryStream(CachedZipFile zipFile, string entryName, string fileName)
         {
-            ZipEntry entry;
+            ZipEntryCache entryCache;
             if (entryName == null)
             {
-                entry = zipFile.EntryCache.FirstOrDefault().Value;
+                entryCache = zipFile.EntryCache.FirstOrDefault().Value;
             }
             else
             {
-                zipFile.EntryCache.TryGetValue(entryName, out entry);
+                zipFile.EntryCache.TryGetValue(entryName, out entryCache);
+            }
+
+            if (entryCache is { Modified: true })
```

Comment:
🔥, Seems like this should address the issue I was having where we were attempting to read an entry after a `Store()`